nexus/types/src/external_api/params.rs

```diff
@@ -1220,13 +1225,25 @@ pub struct VpcSubnetCreate {
     /// be assigned if one is not provided. It must not overlap with any
     /// existing subnet in the VPC.
     pub ipv6_block: Option<Ipv6Net>,
+
+    /// An optional router, used to direct packets sent from hosts in this subnet
+    /// to any destination address.
+    ///
+    /// Custom routers apply in addition to the VPC-wide *system* router, and have
+    /// higher priority than the system router for an otherwise
+    /// equal-prefix-length match.
+    pub custom_router: Option<NameOrId>,
```

Comment:
I was wondering if this could just be called `router` and then on the `VpcSubnet` view we can call it `router_id`, and we always include the ID. But I see now, "Custom routers apply in addition to the VPC-wide *system* router" means that doesn't make sense, right? Because when there is a custom router, there is still also the system router. It's not a default vs. custom situation. So I think this is good.